W-18940999-merge-MS2025-06-ACB-to-latest-mk (#474)
mulesoft/docs-code-builder: modules/ROOT/pages/troubleshoot-gen-ai.adoc

= Troubleshoot Generative AI for Anypoint Code Builder

The following errors sometimes occur when you use Generative AI for Anypoint Code Builder.

> == Unexpected Query Results When Using the Generative Flow Feature

Reviewer: If this only applies to the generative flow feature, this section can be removed. If it applies to the API spec feature, we'll want to revise it to reference that feature only.

Author: Let's put this refactoring on ACB backlog.

If the generative flow doesn't return accurate and useful results for the messages you enter, rewrite your message and follow the guidelines to xref:int-ai-create-integrations.adoc#craft-ai-prompt[craft an effective message].

== Troubleshoot Access Errors

If you can't access generative AI, contact your Anypoint Platform organization administrator. If you're the organization administrator, check these considerations:

* You accepted the terms and conditions and xref:access-management::enabling-einstein.adoc[enabled Einstein for Anypoint Code Builder] for your organization. 

> * The Salesforce organization you are connected to for  requests has Generative AI enabled. For more information, see https://help.salesforce.com/s/articleView?id=sf.generative_ai_enable.htm[Set Up Einstein Generative AI].

Reviewer: ```suggestion
* The Salesforce organization you are connected to for requests has generative AI enabled. For more information, see https://help.salesforce.com/s/articleView?id=sf.generative_ai_enable.htm[Set Up Einstein Generative AI].
```

* The Salesforce organization you are connected to for requests has an xref:access-management::trusted-salesforce-org.adoc[established a tenant relationship] with your Anypoint Platform organization, and the connection is enabled.  
* The Salesforce organization you are connected to for requests is valid. For example, verify that it's not expired or otherwise disabled. 

== Troubleshoot Anypoint Platform Authorization Errors

If you receive an authorization error, contact your Anypoint Platform organization administrator. If you are the organization administrator, check the following:

* The user has the *Mule Developer Generative AI User* permission assigned in Anypoint Platform access management.
* You accepted the terms and conditions to xref:access-management::enabling-einstein.adoc[enable Einstein] for your organization.
* Your Anypoint Platform organization is current and active. 

> == Generative AI Doesn't Appear for EU Cloud Users on the Desktop IDE

Reviewer: ```suggestion
== Generative AI Doesn't Show for EU Cloud Users on the Desktop IDE
```

> If you are signed in to your Anypoint Platform account on the EU Cloud and Generative AI doesn't appear in the UI, verify the following:

Reviewer: ```suggestion
If you are signed in to your Anypoint Platform account on the EU Cloud and generative AI doesn't show in the UI, verify these are correct:
```

* You completed the xref:start-acb.adoc#change-clouds[Set the Desktop IDE to the EU or US Cloud] procedure.
* You are signed in to Anypoint Platform through the Anypoint Code Builder app.

> * xref:int-ai-create-integrations.adoc#before-you-begin[You have the required permissions].

Reviewer: ```suggestion
* xref:api-ai-create-spec.adoc#before-you-begin[You have the required permissions].
```

== Troubleshoot Chat Availability 

Generative AI requires that a supported file is open in the editor when you open the panel. If this condition isn’t met, you might see one of the following messages:

1. "To start a new chat for API, open a compatible API specification." +
This message appears when you're in an API project but the active file is either not an API specification or is written in an unsupported format.
To resolve this issue, open an API spec file in RAML 1.0 or OAS 3.0 format.

2. "To start a new chat for Flow, open an integration project." +
This message appears when you're in an API project or in an integration project but no .xml file is selected.
To fix it, open a .xml file that is part of an integration project.

> If you're in an integration project and have an API spec open, the panel will appear, but you won't be able to generate APIs.

Reviewer: ```suggestion
If you're in an integration project and have an API spec open, the panel shows, but you won't be able to generate APIs.
```

> In this case, you can only use Generative AI to generate Flows. To generate APIs, switch to an API project and open a supported API specification file.

Reviewer: ```suggestion
In this case, you can only use generative AI to generate Flows. To generate APIs, switch to an API project and open a supported API specification file.
```

> If the file you have open is in OpenAPI 2.0 or another unsupported format, you will not be able to use Generative AI. Open a file that uses RAML 1.0 or OAS 3.0 instead.

Reviewer: ```suggestion
If the file you have open is in OpenAPI 2.0 or another unsupported format, you can't use generative AI. Open a file that uses RAML 1.0 or OAS 3.0 instead.
```

> Generative AIa determines the output format based on the file that was in focus when the panel was opened. If a RAML 1.0 file is active, the generated content will be in RAML 1.0. If an OAS 3.0 file is active, the generated content will be in OAS 3.0.

Reviewer: ```suggestion
Generative AI determines the output format based on the file that was in focus when the panel was opened. If a RAML 1.0 file is active, the generated content will be in RAML 1.0. If an OAS 3.0 file is active, the generated content will be in OAS 3.0.
```
